Examples/ObjectiveC/BlockingGesturesDelegateExample.m

```diff
@@ -0,0 +1,56 @@
+#import "BlockingGesturesDelegateExample.h"
+@import Mapbox;
+
+NSString *const MBXExampleBlockingGesturesDelegate = @"BlockingGesturesDelegateExample";
+
+@interface BlockingGesturesDelegateExample () <MGLMapViewDelegate>
+@property (nonatomic) MGLCoordinateBounds colorado;
+@end
+
+@implementation BlockingGesturesDelegateExample
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    MGLMapView *mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds];
+    mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    mapView.rotateEnabled = NO;
+    mapView.minimumZoomLevel = 10;
+    mapView.delegate = self;
+    
+    mapView.styleURL = [MGLStyle outdoorsStyleURLWithVersion:9];
+    
+    // Denver, Colorado
+    CLLocationCoordinate2D center = CLLocationCoordinate2DMake(39.748947, -104.995882);
+    
+    // Starting point.
+    [mapView setCenterCoordinate:center zoomLevel:10 direction:0 animated:NO];
+    
+    // Colorado's bounds
+    CLLocationCoordinate2D ne = CLLocationCoordinate2DMake(40.989329, -102.062592);
+    CLLocationCoordinate2D sw = CLLocationCoordinate2DMake(36.986207, -109.049896);
+    self.colorado = MGLCoordinateBoundsMake(sw, ne);
+    
+    [self.view addSubview:mapView];
+}
+
+- (BOOL)mapView:(MGLMapView *)mapView shouldChangeFromCamera:(MGLMapCamera *)oldCamera toCamera:(MGLMapCamera *)newCamera
+{
+    // Get current coordinates
+    CLLocationCoordinate2D newCameraCenter = newCamera.centerCoordinate;
+    MGLMapCamera *camera = mapView.camera;
+    
+    // Get new bounds
```

Comment:
These comments say _what_ the developer is expected to do, but not _why_. An example needs to explain the rationale behind each non-obvious step, in order to avoid the phenomenon of [cargo cult programming](http://www.catb.org/~esr/jargon/html/C/cargo-cult-programming.html). In particular, we should be explicit about the fact that we’re changing and resetting the map view’s camera in order to predict the extents of the new viewport.